sdk/src/account_info.rs

```diff
@@ -0,0 +1,304 @@
+    /// Lamports.
+    pub lamports: Option<u64>,
+    /// Discriminator.
+    pub discriminator: Option<[u8; 8]>,
+    /// Account data.
+    pub data: Option<Rc<RefCell<Vec<u8>>>>,
+    /// Data hash.
+    pub data_hash: Option<[u8; 32]>,
+    /// Address.
+    pub address: Option<[u8; 32]>,
+    /// New Merkle tree index. Set only if you want to change the tree.
+    pub output_merkle_tree_index: Option<u8>,
+    /// New address parameters.
+    pub new_address_params: Option<PackedNewAddressParams>,
+}
+
+impl<'a> LightAccountInfo<'a> {
+    pub fn from_meta_init(
+        meta: &'a LightAccountMeta,
+        discriminator: [u8; 8],
+        new_address: [u8; 32],
+        new_address_seed: [u8; 32],
+        program_id: &Pubkey,
+    ) -> Result<Self> {
+        let address_merkle_context = meta
+            .address_merkle_context
+            .as_ref()
+            .ok_or(LightSdkError::ExpectedAddressMerkleContext)?;
+
+        let new_address_params = PackedNewAddressParams {
+            seed: new_address_seed,
+            address_queue_account_index: address_merkle_context.address_queue_pubkey_index,
+            address_merkle_tree_account_index: address_merkle_context
+                .address_merkle_tree_pubkey_index,
+            address_merkle_tree_root_index: meta
+                .address_merkle_tree_root_index
+                .ok_or(LightSdkError::ExpectedAddressRootIndex)?,
+        };
+
+        let account_info = LightAccountInfo {
+            input: None,
+            owner: Some(*program_id),
+            // Needs to be assigned by the program.
+            lamports: None,
+            // Needs to be assigned by the program.
+            discriminator: Some(discriminator),
+            // NOTE(vadorovsky): A `clone()` here is unavoidable.
+            // What we have here is an immutable reference to `LightAccountMeta`,
+            // from which we can take an immutable reference to `data`.
+            //
+            // - That immutable reference can be used in the input account,
+            //   since we don't make modifications there.
+            // - In the most cases, we intend to make modifications for the
+            //   output account. We make a copy, which then we try not to
+            //   copy again until the moment of creating a CPI call.
+            //
+            // The reason why `solana_account_info::AccountInfo` stores data as
+            // `Rc<RefCell<&'a mut [u8]>>` is that the reference points to
+            // runtime's memory region which provides the accout and is mutable
+            // by design.
+            //
+            // In our case, compressed accounts are part of instruction data.
+            // Instruction data is immutable (`&[u8]`). There is no way to
+            // mutate instruction data without copy.
+            data: meta
+                .data
+                .as_ref()
+                .map(|data| Rc::new(RefCell::new(data.clone()))),
```

Comment:
shouldn't this just be a new vector since it is an empty account?
```suggestion
            data: Vec::new(),
```

Reply:
Maybe it makes sense to init with capacity?
```suggestion
            data: Rc::new(RefCell::new(Vec::with_capacity(capacity.unwrap_or_default()))),
```

Reply:
Done, thanks!